pkg/controller/instanceset/instance_util.go

```diff
@@ -236,16 +258,55 @@ func GenerateAllInstanceNames(parentName string, replicas int32, templates []Ins
 	return instanceNameList
 }
 
+func GenerateAllInstanceNamesWithOrdinals(parentName string, replicas int32, templates []InstanceTemplate, offlineInstances []string, defaultTemplateOrdinals workloads.Ordinals) ([]string, error) {
+	totalReplicas := int32(0)
+	instanceNameList := make([]string, 0)
+	for _, template := range templates {
+		replicas := template.GetReplicas()
+		ordinalList, err := ConvertOrdinalsToOrdinalList(template.GetOrdinals())
+		if err != nil {
+			return nil, err
+		}
+		names, err := GenerateInstanceNamesFromTemplateWithOrdinalList(parentName, template.GetName(), replicas, offlineInstances, ordinalList)
+		if err != nil {
+			return nil, err
+		}
+		instanceNameList = append(instanceNameList, names...)
+		totalReplicas += replicas
+	}
+	if totalReplicas < replicas {
+		ordinalList, err := ConvertOrdinalsToOrdinalList(defaultTemplateOrdinals)
+		if err != nil {
+			return nil, err
+		}
+		names, err := GenerateInstanceNamesFromTemplateWithOrdinalList(parentName, "", replicas-totalReplicas, offlineInstances, ordinalList)
+		if err != nil {
+			return nil, err
+		}
+		instanceNameList = append(instanceNameList, names...)
+	}
+	getNameNOrdinalFunc := func(i int) (string, int) {
+		return ParseParentNameAndOrdinal(instanceNameList[i])
+	}
+	baseSort(instanceNameList, getNameNOrdinalFunc, nil, true)
+	return instanceNameList, nil
+}
+
 func GenerateInstanceNamesFromTemplate(parentName, templateName string, replicas int32, offlineInstances []string) []string {
-	instanceNames, _ := GenerateInstanceNames(parentName, templateName, replicas, 0, offlineInstances)
+	instanceNames := GenerateInstanceNames(parentName, templateName, replicas, 0, offlineInstances)
 	return instanceNames
 }
 
+func GenerateInstanceNamesFromTemplateWithOrdinalList(parentName, templateName string, replicas int32, offlineInstances []string, ordinalList []int32) ([]string, error) {
```

Comment:
The same with GenerateInstanceNamesFromTemplate and GenerateInstanceNamesFromTemplateWithOrdinalList

Reply:
You must ensure that all callers receive the correct list of pod names, regardless of whether the ordinals parameter is set or not.